src/PSRule.Rules.Azure/rules/Azure.Template.Rule.ps1

```diff
@@ -304,10 +304,64 @@ Rule 'Azure.Template.ValidSecretRef' -Type '.json' -If { (IsParameterFile) } -Ta
     }
 }
 
+# Synopsis: Use comments for each resource in ARM template to communicate purpose.
+Rule 'Azure.Template.UseComments' -Type '.json' -If { (IsTemplateFile) -and !(IsGenerated) } -Tag @{ release = 'GA'; ruleSet = '2021_12'; } {
+    $resources = @(GetTemplateResources);
```

Comment:
What about?

```suggestion
    $resources = @(GetTemplateResources | Where-Object { $Assert.NullOrEmpty($_, "comments").Result });
    $Assert.Count($resources, '.', 0).Reason(
        $LocalizedData.TemplateResourceWithoutComment,
        $TargetObject.FullName,
        $numResourcesWithoutComments
      );
```

Reply:
@BernieWhite Nice, this looks much better. I'll do the same for `Azure.Template.UseDescriptions`.